pkg/operator/controller/ingress/controller.go

```diff
@@ -548,6 +548,7 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, platformStat
 					changed = true
 				}
 				if statusLB.ProviderParameters.AWS.Type == operatorv1.AWSClassicLoadBalancer {
+					statusLB.ProviderParameters.AWS.NetworkLoadBalancerParameters = nil
```

Comment:
I see that the return value of `setDefaultPublishingStrategy` is not really taken into account by the calling function (why btw) but at least the unit test does. 1. should we set `changed=true` taken into account that we reset the status? 2. I think that we should add unit tests to see the status gets reset.

Reply:
Yea I find this pattern a bit odd. `changed` is only used in unit testing. The really odd thing is: why do you even need a `changed` bool for unit testing? Our unit test `TestSetDefaultPublishingStrategyHandlesUpdates` diffs the expected IngressController with the actual, so we are measuring the "change" that way. Seems much better to detect the change through the actual diff, and not have an artificial `change` flag, which a bit meaningless, prone to error, and arbitrary (I don't think it adds any value).

I think the pattern with `changed` the original author was trying to follow was similar to [recordIngressCondition](https://github.com/openshift/router/blob/fa20a2437f013dfc51d360e3f50b43df1db590bc/pkg/router/controller/status.go#L197), where `changed` is actually used to indicate if an update is needed. However, that isn't the case with `setDefaultPublishingStrategy`, as we always update the status in `admit`.

> 1. should we set changed=true taken into account that we reset the status?

Given that:
1. These [initialization-type](https://github.com/openshift/cluster-ingress-operator/blob/7edef371026129f765a86748148c79b38fba636e/pkg/operator/controller/ingress/controller.go#L537-L545) changes don't set `change=true` when the status is actually changed (I think this PR's status "clean up" is similar to these status "initialization")
2. My point above about `changed` having little to no value and it appears like it's not being maintained properly (shouldn't `changed` be true when anything in the status changes?)

I vote to not worry about it.

> 2. I think that we should add unit tests to see the status gets reset.

Ah I completely forgot about unit testing. Done.